pkg/cloud/services/loadbalancer/loadbalancer.go

```diff
@@ -153,71 +115,186 @@ func (s *Service) ReconcileLoadBalancer(openStackCluster *infrav1.OpenStackClust
 		}
 	}
 
-	allowedCIDRs := []string{}
-	// To reduce API calls towards OpenStack API, let's handle the CIDR support verification for all Ports only once.
-	allowedCIDRsSupported := false
-	octaviaVersions, err := s.loadbalancerClient.ListOctaviaVersions()
+	allowedCIDRsSupported, err := s.isAllowsCIDRSSupported(lb)
 	if err != nil {
 		return false, err
 	}
-	// The current version is always the last one in the list.
-	octaviaVersion := octaviaVersions[len(octaviaVersions)-1].ID
-	if openstackutil.IsOctaviaFeatureSupported(octaviaVersion, openstackutil.OctaviaFeatureVIPACL, lbProvider) {
-		allowedCIDRsSupported = true
+
+	// AllowedCIDRs will be nil if allowed CIDRs is not supported by the Octavia provider
+	if allowedCIDRsSupported {
+		lbStatus.AllowedCIDRs = getCanonicalAllowedCIDRs(openStackCluster)
+	} else {
+		lbStatus.AllowedCIDRs = nil
 	}
 
 	portList := []int{apiServerPort}
 	portList = append(portList, openStackCluster.Spec.APIServerLoadBalancer.AdditionalPorts...)
 	for _, port := range portList {
-		lbPortObjectsName := fmt.Sprintf("%s-%d", loadBalancerName, port)
+		if err := s.reconcileAPILoadBalancerListener(lb, openStackCluster, clusterName, port); err != nil {
+			return false, err
+		}
+	}
 
-		listener, err := s.getOrCreateListener(openStackCluster, lbPortObjectsName, lb.ID, port)
+	return false, nil
+}
+
+// getAPIServerVIPAddress gets the VIP address for the API server from wherever it is specified.
+// Returns an empty string if the VIP address is not specified and it should be allocated automatically.
+func getAPIServerVIPAddress(openStackCluster *infrav1.OpenStackCluster) (string, error) {
```

Comment:
we could unit test it but I'm not going to nitpick on that. Just make sure the existing tests cover the output of that function and I'm fine.

Reply:
I wrote it.